### PR TITLE
Mark post as read after viewing the image or voting

### PR DESCRIPTION
**Pull request by ktgd**

### CODE REVIEW

File: components/hooks/feeds/useFeedItem.ts [hunk 1]
```diff
@@ -89,26 +91,30 @@ const useFeedItem = (
...
-        );
-      }
+      setPostRead();
     }
 
     dispatch(setPost(post));
     navigation.push("Post");
   };
 
+  const setPostRead = () => {
```
Comment (gkasdorf) [inline components/hooks/feeds/useFeedItem.ts:101]: added this to mark the posts as read in the state

File: components/hooks/feeds/useFeedItem.ts [hunk 2]
```diff
@@ -49,6 +49,14 @@ const useFeedItem = (
     );
 
     try {
+      lemmyInstance.markPostAsRead({
```
Comment (gkasdorf) [inline components/hooks/feeds/useFeedItem.ts:54]: use `setPostRead` to update the post in state

File: components/ui/ContentView.tsx [hunk 3]
```diff
@@ -75,6 +80,13 @@ function ContentView({ post, isPreview = false, recycled }: ContentViewProps) {
 
   const onImagePress = () => {
```
Comment (gkasdorf) [inline components/ui/ContentView.tsx:81]: use `feedItem.setPostRead()` from `FeedItem.tsx`. Should just pass the function into ContentView with a prop.

File: components/ui/Feed/CompactFeedItem/CompactFeedItemThumbnail.tsx [hunk 4]
```diff
@@ -23,11 +27,19 @@ function CompactFeedItemThumbnail({
   imageViewOpen: boolean;
 }) {
   const theme = useTheme();
+  const dispatch = useAppDispatch();
 
   const { blurNsfw } = useAppSelector(selectSettings);
 
   const onImagePress = () => {
```
Comment (gkasdorf) [inline components/ui/Feed/CompactFeedItem/CompactFeedItemThumbnail.tsx:34]: Same thing here. Just use `feedItem.setPostRead()`

**Review by gkasdorf — CHANGES_REQUESTED** [review]
Mark the post as read in state. I have moved the logic in the on post press handler to a separate function, so you can call that where needed.

**Comment by gkasdorf** [comment]
Like I said in the reviews, we need to make sure the state gets updated whenever we perform those actions. Otherwise it will only reflect on Lemmy and you'll need to refresh for the changes to be visible.

Also, I added a screen in Settings for you. Can we fill this out and have different settings for these? Some people don't seem to want image viewing to equate to reading, but others do. So I think it would be a good idea for this to be a toggle. Probably with these settings:

1. Post press is read
2. Image view is read
3. Post vote is read

All you need to add in `settingsSlice.tsx` is the options to `SettingsState` interface (i.e. `markReadOn:boolean`) and add the initial setting to the initial state. This will automatically update user settings on next load with the new values.

If you have questions let me know. I'd work on it myself but have other things I am working on right now.

Thanks!

**Comment by ktgd** [comment]
Thanks for the tips and suggestions. I'll work on implementing the changes tomorrow!

**Comment by ktgd** [comment]
Those changes have been implemented and I added options to the settings for:
- Mark Read on Post Open
- Mark Read on Image View
- Mark Read on Vote

Everything is working for the Compact view, but Post Open and Vote aren't being triggered on the Normal view. I will look into that tomorrow. 

Thanks!

**Review by gkasdorf — APPROVED** [review]

**Comment by gkasdorf** [comment]
Everything looks good now! Made a [a few changes](https://github.com/gkasdorf/memmy/pull/202/commits/1eec15a66ebae81fc7fdf365c1c74401cca3fa2e) to how it performs the update when liking the post (should make the like post request before the read post request). Also added post.read to the memo, but that was my fault not yours.

Thanks for this one!

**Pull request merged**